Doc/library/dataclasses.rst

```diff
@@ -491,6 +491,25 @@ depend on one or more other fields.  For example::
         def __post_init__(self):
             self.c = self.a + self.b
 
+The :meth:`__init__` method generated by :func:`dataclass` does not call base class
+:meth:`__init__` methods. If the base class has an :meth:`__init__` method that has to be called,
+it is common to call this method in a :meth:`__post_init__` method::
+
+    @dataclass
+    class Rectangle:
+        height: float
+        width: float
+
+    @dataclass
+    class Square(Rectangle):
+        side: float
+
+        def __post_init__(self):
+            super().__init__(self.side, self.side)
+
+Note, however, that in general the dataclass-generated :meth:`__init__` methods not need to be
+called, since the derived class will take care of initializing all fields.
+
```

Comment:
This looks good. How about instead of "the derived class", use "the derived dataclass"? I'm trying to draw the distinction that if a non-dataclass is derived from a dataclass, then the base class's `__init__` should in fact be called.

Reply:
If I understand PEP 557 correctly, the `__init__` method needs to be called if a dataclass inherits from a non-dataclass or if a non-dataclass inherits from a dataclass. I have tried to reflect this in the new wording.

Reply:
> If I understand PEP 557 correctly, the `__init__` method needs to be called if a dataclass inherits from a non-dataclass or if a non-dataclass inherits from a dataclass. I have tried to reflect this in the new wording.

If a dataclass inherits from a non-dataclass, then it's probably that the base non-dataclass's `__init__` **does** need to be called. That's what `__post_init__` is for, among other uses.

If a non-dataclass inherits from a dataclass, then the non-dataclass most likely does need to call the dataclass's `__init__`.

It's only the case where a dataclass inherits from a dataclass that the base class's `__init__` most likely doesn't need to be called. I say "most likely" because you could construct examples where it does need to be called, if the base dataclass's `__post_init__` does some processing. But it's probably too complex to mention that here, I'm just mentioning it here to help you understand the issue. Apologies if I'm not being helpful.

Reply:
This is very helpful! Thank you!